chia/full_node/full_node.py

```diff
@@ -1615,8 +1612,14 @@ async def add_prevalidated_blocks(
                     vs.difficulty = cc_sub_slot.new_difficulty
                     assert expected_sub_slot_iters == vs.ssi
                     assert expected_difficulty == vs.difficulty
+            block_rec = blockchain.block_record(block.header_hash)
```

Comment:
clever. I had expected that we would have to pass along this block along the call path. But this makes sense. This throws if the block record is missing, right?

Reply:
doesn't throw but this block should (as far as i can think of) always be present, it will return None and then behave as before